readme.md

```diff
@@ -0,0 +1,177 @@
+        #       node_purpose : object containing name and type of node:
+        #                      name: Short description of the node
+        #                      type: integer that should reflect the types of the nodes 
+        #                            at node_input_factory/node_input_enums.
+        #       For example the node_purpose type below is 0 and that is reflected with a
+        #       DistanceNode.
+
+        nodes: [{local: false, eds_location: 'eds_files/Arduino1.eds', node_purpose: {name: 'Front view object distance', type: 0}}]
+        
+        # Sets the max speed of wheelNodes.
+        max_speed_in_ms: 3 
+        
+        # Object related to can data logging.
+        # Our logger makes use of a buffered logger implementation.
+
+        raw_can_data_logging: {
+            # Enables of Disables the buffered logger.
+            enabled: true,
+
+            # The size of the buffer so with it can contain 1024 can messages.
+            buffer: 1024
+        }
+```
+## Extending CargoDash Communication Possibilities
+
+For now, CargoDash is only able to work with the CanOpen protocol but it has the potential to work with more communication protocols. If you want to add a communication protocol then you should add a new interpreter and a new listener. You can easily create a new listener class that uses the same methods as the abstract listener class. The same goes for the interpreter. It's important that you create nodes from the NodeFactory inside your interpreter. The NodeFactory class creates 'Node Input' objects which can be used to change the vehicle's state. In the diagram below you can have an understanding of CargoDash internal class interaction works.
+
+
+![CargoDash Architecture](img/api_cargodash_v6.png "CargoDash Architecture")
+
+## Adding Nodes
+In CargoDash we already implemented certain standard nodes that would exist in a real vehicle such as SteeringNode, TemperatureNode, and more. But if you want to add your own, custom node you can add it by changing some files. You can do this using the changes described below using a LidarNode as an example.
+
+### node_input_factory/node_input_enums.py
+
+Just add an enum to the enums list.
+```python
+    ...
+    LidarNode=5
+```
+### node_input_factory/node_input_classes.py
+
+At the end of the file add the new class *LidarNodeInput*.
+```python
+    ...
+    class LidarNodeInput(NodeInput):
+    # Enum 5
+    def __init__(self, distance: float, name: str, node_name: str):
+        super().__init__(distance, name, node_name)
+
+```
+### node_input_factory/node_input_factory.py
+
+Inside the class *NodeInputFactory* add a new method.
+```python
+    ...
+    def create_lidar_node_input(self, distance: float,
+                                   name: str, node_name: str):
+        return LidarNodeInput(distance,
+                              name, node_name)
+```
+And don't forget to import *LidarNodeInput* at the *node_input_factory.py* file.
```

Comment:
persoonlijk vind ik het mooier om bestandsnamen in een `code snippet` te zetten. Kijk maar of je het verandert of niet